From df2a397d58626741f1bb33c6ee565d516d9dbc18 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Fri, 22 Aug 2025 18:17:21 +0800
Subject: [PATCH 116/123] mtk: mt76: mt7996: fix binfile precal issue

Add the missing return when when successfully reading precal data from
the binfile

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 mt7996/eeprom.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7996/eeprom.c b/mt7996/eeprom.c
index 7a5821ff..612ff3f7 100644
--- a/mt7996/eeprom.c
+++ b/mt7996/eeprom.c
@@ -576,6 +576,7 @@ static int mt7996_eeprom_load_precal(struct mt7996_dev *dev)
 		ret = mt7996_eeprom_load_precal_binfile(dev, offs, size);
 		if (ret)
 			goto fail;
+		return 0;
 	}
 
 	ret = mt76_get_of_data_from_mtd(mdev, dev->cal, offs, size);
-- 
2.45.2

